### ISSUE TITLE
[Feature]: Consider activating Rust integer overflow protection

### ISSUE THREAD

**Issue by evonide** [stacks-network/sbtc#1173]
# Feature - Consider activating Rust integer overflow protection

## 1. Description

We should measure the impact of activating Rust integer overflow checks as per https://doc.rust-lang.org/cargo/reference/profiles.html#overflow-checks. This can be activated through a config change in https://github.com/stacks-network/sbtc/blob/main/Cargo.toml.

It's recommended to activate this if the resulting performance tradeoff can be accepted.

Note: We should ensure this will cover WSTS as well.

**Comment by djordon** [comment]
Oh I thought that it panicked by default, but it appears that it wraps by default. Yeah we should do this.

**Comment by djordon** [comment]
> [@djordon](https://github.com/djordon) do we have any benchmarks to measure performance after this changes and make a decision if it is acceptable?

No, but it is almost certainly fine. We can go ahead and do this without the benchmark, since it's very unlikely to have a noticeable impact on performance. Plus we would notice anything unacceptable on testnet or local devnet and can easily revert.
